### PR TITLE
EDM-922: Fine-tune HTTP server settings

### PR DESCRIPTION
**Pull request by avishayt**
Also collect logs in smoke/e2e github workflows

## Summary by CodeRabbit

- **New Features**
  - Introduced a GitHub Action for collecting logs from Kubernetes pods and uploading them as artifacts.
  - Added new steps in E2E and smoke testing workflows to collect and upload logs post-testing.
  - Enhanced API configuration with new HTTP timeout and limit parameters.

- **Bug Fixes**
  - Improved request handling by adding middleware for request size and header limits.

- **Documentation**
  - Updated configuration structures to include new HTTP-related settings and improved JSON handling for duration values.

### CODE REVIEW

**Comment by romfreiman** [comment]
@coderabbitai review

File: .github/actions/collect-logs/action.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+             > ${{ inputs.log-directory }}/${statefulset}_${timestamp}.log; then
+            log_files_collected=true
+          else
+            echo "Error collecting logs for $statefulset"
+          fi
+        done
+
+        if [[ "$log_files_collected" == false ]]; then
+          echo "No log files were collected."
+        fi
```
[the remark this answers is not in the document]
Comment (avishayt) [reply .github/actions/collect-logs/action.yml:57]: What do you mean by "The steps.filter.outputs.notdocs reference suggests a missing filter step"?

File: internal/config/config.go [hunk 2]
```diff
@@ -33,20 +35,27 @@ type dbConfig struct {
...
+	SrvKeyFile            string   `json:"srvKeyFile,omitempty"`
+	AltNames              []string `json:"altNames,omitempty"`
+	LogLevel              string   `json:"logLevel,omitempty"`
+	HttpReadTimeout       Duration `json:"httpReadTimeout,omitempty"`
+	HttpReadHeaderTimeout Duration `json:"httpReadHeaderTimeout,omitempty"`
+	HttpWriteTimeout      Duration `json:"httpWriteTimeout,omitempty"`
+	HttpMaxNumHeaders     int      `json:"httpMaxNumHeaders,omitempty"`
+	HttpMaxHeaderBytes    int      `json:"httpMaxHeaderBytes,omitempty"`
+	HttpMaxUrlLength      int      `json:"httpMaxUrlLength,omitempty"`
+	HttpMaxRequestSize    int      `json:"httpMaxRequestSize,omitempty"`
```
Comment (fzdarsky) [inline internal/config/config.go:58]: It'd be awesome to have a bit more structure to the svcConfig and I was wondering whether user- and agent-facing servers would need different timeouts, but that can well be in a follow-up PR.

**Review by fzdarsky — APPROVED** [review]

File: internal/config/config.go [hunk 3]
```diff
@@ -75,6 +84,35 @@ type prometheusConfig struct {
...
+		d.Duration, err = time.ParseDuration(value)
+		if err != nil {
+			return err
+		}
+		return nil
+	default:
+		return errors.New("invalid duration")
+	}
+}
+
```
[the remark this answers is not in the document]
Comment (avishayt) [reply internal/config/config.go]: Fixed!

File: internal/config/config.go (hunk 2 above)
Comment (avishayt) [reply internal/config/config.go]: Sure, let's discuss in a folluw-up

**Review by asafbss — APPROVED** [review]

**Pull request merged**